### PR TITLE
[Event Hubs] Handle cases where desired worker count exceeds int.MaxInt

### PR DESCRIPTION
**Pull request by chiangvincent**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

If the worker count exceeds int.MaxInt, we automatically return the last value in the sorted worker count array, since the sortedWorkerCount only contains ints (max partition count in EventHub is 1000).

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs [hunk 1]
```diff
@@ -163,9 +163,15 @@ internal static int[] GetSortedValidWorkerCountsForPartitionCount(int partitionC
         /// <param name="workerCount">The value that we want to find in the sortedValues list (if it exists), or the next largest element.</param>
         /// <param name="sortedValidWorkerCountList">The list of valid worker counts. This must be sorted.</param>
         /// <returns></returns>
-        internal static int GetValidWorkerCount(int workerCount, int[] sortedValidWorkerCountList)
+        internal static int GetValidWorkerCount(long workerCount, int[] sortedValidWorkerCountList)
```
Comment (alrod) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs:166]: Can we add more detailed comment why do we need this method.

**Review by jsquire — APPROVED** [review]

**Review by alrod — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ internal static TargetScalerResult ThrottleScaleDownIfNecessaryInternal(TargetSc
         internal TargetScalerResult GetScaleResultInternal(TargetScalerContext context, long eventCount, int partitionCount)
         {
             int desiredConcurrency = GetDesiredConcurrencyInternal(context);
-            int desiredWorkerCount = (int)Math.Ceiling(eventCount / (decimal)desiredConcurrency);
+            long desiredWorkerCount = (long)Math.Ceiling(eventCount / (decimal)desiredConcurrency);
```
Comment (pragnagopa) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs:89]: should we think about enforcing a max on the desired worker count == int.max

Comment (chiangvincent) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs:89]: in this case, we are effectively doing the same thing 

the largest target worker count for event hub is the last element in the sortedValidWorkerCountList (which is the partition count) -- see `GetValidWorkerCount`

**Review by pragnagopa — APPROVED** [review]

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs (hunk 2 above)
Comment (paulbatum) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs:89]: whats the advantage of doing it this way? Its a bit harder to understand than doing a math.max

Comment (chiangvincent) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsTargetScaler.cs:89]: logically there's no difference, I suppose to enforce the max approach, we'd have to save an extra variable long desiredWorkerCount to compare against maxInt. We'd then use this variable to pass into `GetValidWorkerCount`

**Review by paulbatum — APPROVED** [review]

**Pull request merged**